src/codec/custom/BitmapIndexOptionsCodec.ts

```diff
@@ -19,36 +19,41 @@ import {FixSizedTypesCodec} from '../builtin/FixSizedTypesCodec';
 import {BitsUtil} from '../../BitsUtil';
 import {ClientMessage, BEGIN_FRAME, END_FRAME, Frame, DEFAULT_FLAGS} from '../../ClientMessage';
 import {CodecUtil} from '../builtin/CodecUtil';
-import {BitmapIndexOptions} from '../../config/BitmapIndexOptions';
 import {StringCodec} from '../builtin/StringCodec';
+import {
+    uniqueKeyTransformationToId,
+    uniqueKeyTransformationFromId,
+    BitmapIndexOptionsImpl,
+} from '../../config/BitmapIndexOptions';
 
 const UNIQUE_KEY_TRANSFORMATION_OFFSET = 0;
 const INITIAL_FRAME_SIZE = UNIQUE_KEY_TRANSFORMATION_OFFSET + BitsUtil.INT_SIZE_IN_BYTES;
 
 export class BitmapIndexOptionsCodec {
-    static encode(clientMessage: ClientMessage, bitmapIndexOptions: BitmapIndexOptions): void {
+    static encode(clientMessage: ClientMessage, bitmapIndexOptions: BitmapIndexOptionsImpl): void {
         clientMessage.addFrame(BEGIN_FRAME.copy());
 
         const initialFrame = Frame.createInitialFrame(INITIAL_FRAME_SIZE, DEFAULT_FLAGS);
-        FixSizedTypesCodec.encodeInt(initialFrame.content, UNIQUE_KEY_TRANSFORMATION_OFFSET, bitmapIndexOptions.uniqueKeyTransformation);
+        const transformationId = uniqueKeyTransformationToId(bitmapIndexOptions.uniqueKeyTransformation);
```

Comment:
I'm thinking of converting `BitmapIndexOptionsCodec` and `IndexConfigCodec` codecs into builtin codecs, so that they're not generated. Do you think it's a bad idea to do so?

Reply:
Implement it in a way that doesn't require changes in codecs. The only change to be made is in https://github.com/hazelcast/hazelcast-client-protocol/pull/333